### PR TITLE
fix inplace_update for daemonset and support inplace

### PR DESCRIPTION
**Pull request by Noisyes**
Signed-off-by: kongyufeng <noisyes@B-R7NSMD6R-0224.local>

### Ⅰ. Describe what this PR does
daemonset inplace update strategy  implementation has problems. Should refresh update expectations for new pod rather than old pod.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
add e2e test to check old pod and new pod is same.

### Ⅳ. Describe how to verify it
e2e test pass

### CODE REVIEW

**Comment by FillZpp** [comment]
/hold

Waiting for #711 to be merged first.

**Comment by jetmuffin** [comment]
@FillZpp seems that #711 has been merged and we can start to review this PR now?

**Comment by FillZpp** [comment]
> @FillZpp seems that #711 has been merged and we can start to review this PR now?

Sure, please rebase upstream/master and push your branch again. @Noisyes

File: pkg/util/inplaceupdate/inplace_utils.go [hunk 1]
```diff
@@ -36,7 +39,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var inPlaceUpdatePatchRexp = regexp.MustCompile("^/spec/containers/([0-9]+)/image$")
+var (
+	inPlaceUpdatePatchRexp = regexp.MustCompile("^/spec/containers/([0-9]+)/image$")
+	UpdateExpectations     = expectations.NewUpdateExpectations(&revisionAdapterImpl{})
```
Comment (FillZpp) [inline pkg/util/inplaceupdate/inplace_utils.go:44]: `inplaceupdate` is a public package, don't put `UpdateExpectations`/`revisionAdapterImpl` of daemonset in here.

**Review by FillZpp — COMMENTED** [review]
This PR just enables the `InPlace` strategy for daemonset, but it has not solved all existing problems in the daemonset inplace update logic.

File: pkg/controller/daemonset/daemonset_controller.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ var (
 
 	// A TTLCache of pod creates/deletes each ds expects to see
 	expectations       = kubecontroller.NewControllerExpectations()
-	updateExpectations = kruiseExpectations.NewUpdateExpectations(revisionadapter.NewDefaultImpl())
+	updateExpectations = kruiseExpectations.NewUpdateExpectations(&revisionAdapterImpl{})
```
Comment (FillZpp) [inline pkg/controller/daemonset/daemonset_controller.go:87]: `daemonset` does not need to implement a new `revisionAdapterImpl`, for it has no extra feature like short hash. Just the default impl is enough.

Comment (Noisyes) [reply pkg/controller/daemonset/daemonset_controller.go]: maybe should check it, i remember  use default revision adapter  will cause some problems.

Comment (FillZpp) [reply pkg/controller/daemonset/daemonset_controller.go]: Can you describe the problems you meet?

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**